packages/transform/src/types.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+export type TransformHandler = (frame: object | null) => void;
+export type HandlerRegister = (
+  id: string,
+  handler: TransformHandler
+) => () => void;
+
+export interface HandlersList {
+  [id: string]: TransformHandler[];
+}
+
+export interface TransformsList {
+  [id: string]: Record<string, unknown> | null;
+}
+
+export interface State {
```

Comment:
@swissspidy Do you happen to have a good example of creating a context with actions and state that I could check? Not sure exactly how to do this here, had to use [type casting](https://github.com/GoogleForCreators/web-stories-wp/pull/12202/files#diff-ce14352028d036cff60c103838a120678c7279b6ac26f2a9e83a279ba5582d82R44) which doesn't seem right here. Or is this an example of the `useContextSelector` that might still need work (since it's all related to that, and has to match the typing of the context selector)?

Reply:
So far I only attempted this in `packages/units/src/context.ts` / `packages/units/src/contextProvider.tsx` but yeah it still needs some work to get type support when using `useContextSelector`.

This might do the trick though:

https://github.com/GoogleForCreators/web-stories-wp/tree/try/fix-context-state

This way I get type support when doing something like

```js
useUnits(({ state: { pageSize } }) => pageSize);
```

Reply:
Hm, not sure if I did something incorrectly but I still have to [assign types](https://github.com/GoogleForCreators/web-stories-wp/pull/12202/files#diff-0408b18c76af2834bb09ccf3f2e26c80643b11f5634f017a02d77b314f822e5eR41) in the provider for it not to throw errors.